configure.ac

```diff
@@ -18,6 +18,17 @@ AC_PROG_CC_STDC
 AC_PROG_MAKE_SET
 AM_PROG_AR
 
+# If an mpicc executable is found or tests are disabled, bypass the check for
+# whether --with-mpi= is specified.
+AC_CHECK_PROG(BYPASS_MPICC_CHECK, [mpicc], "yes", "no")
```

Comment:
The first argument should be MPICC, so that you can run:

```
./configure MPICC=/opt/amazon/openmpi/bin/mpicc --prefix=....
```

`AC_CHECK_PROG` is really kind of ugly, because at the end of the day, you actually want to have found the location for mpicc.  I generally prefer to use `AC_CHECK_PROGS`, which has slightly saner semantics.  Something like:

```
AC_CHECK_PROG([MPICC], [mpicc], [no])
```

Finally, I think I'd move this test down to the other MPI tests.  It's kind of weird to have them split by 40 lines of code.